json/json_test.go

```diff
@@ -51,3 +53,52 @@ func testEncParseConsistent(
 			}
 		})
 }
+
+func TestStringEncoding(t *testing.T) {
+	type testCase struct {
+		in         string
+		htmlEscape bool
+		expected   string
+	}
+
+	cases := map[string]testCase{
+		"no escaping required": testCase{
+			in:       "hello world",
+			expected: `"hello world"`,
+		},
+		"json required escaping": testCase{
+			in:         `"hello \nworld"`,
+			htmlEscape: false,
+			expected:   `"\"hello \\nworld\""`,
+		},
+		"html escape with json required escaping": testCase{
+			in:         `"hello \nworld"`,
+			htmlEscape: true,
+			expected:   `"\"hello \\nworld\""`,
+		},
+		"html with html escaping enabled": testCase{
+			in:         "<hello>world</hello>",
```

Comment:
Can you add a test for `<hello id="hola">` and `<hello id='hola'>`, I am scared by quotes.

Reply:
done